mrmustard/physics/fock.py

```diff
@@ -818,6 +818,57 @@ def estimate_quadrature_axis(cutoff, minimum=5, period_resolution=20):
     return xaxis
 
 
+def quadrature_basis(
+    fock_array: Tensor,
+    quad: Batch[Vector],
+    conjugates: bool | list[bool] = False,
+    phi: Scalar = 0.0,
+):
+    r"""Given the Fock basis representation return the quadrature basis representation.
+
+    Args:
+        fock_array (Tensor): fock tensor amplitudes
+        quad (Batch[Vector]): points at which the quadrature basis is evaluated
+        conjugates (list[bool]): which dimensions of the array to conjugate based on
+            whether it is a bra or a ket
+        phi (float): angle of the quadrature basis vector
+
+    Returns:
+        tuple(Tensor): quadrature basis representation at the points in quad
+    """
+    dims = len(fock_array.shape)
+
+    if quad.shape[-1] != dims:
+        raise ValueError(
+            f"Input fock array has dimension {dims} whereas ``quad`` has {quad.shape[-1]}."
+        )
+
+    conjugates = conjugates if isinstance(conjugates, Iterable) else [conjugates] * dims
+
+    # construct quadrature basis vectors
+    cutoffs = fock_array.shape
```

Comment:
I think we need to be careful with how we mean "shape" and "cutoff". Usually cutoff refers to the largest photon number, which is 1 less than the shape.